diff --git a/cmd/go-camo/main.go b/cmd/go-camo/main.go
index cd03faf..9fa8b0b 100644
--- a/cmd/go-camo/main.go
+++ b/cmd/go-camo/main.go
@@ -22,6 +22,8 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/coreos/go-systemd/activation"
 )
 
 const metricNamespace = "camo"
@@ -58,7 +60,7 @@ func main() {
 		SSLKey              string        `long:"ssl-key" description:"ssl private key (key.pem) path"`
 		SSLCert             string        `long:"ssl-cert" description:"ssl cert (cert.pem) path"`
 		AllowList           string        `long:"allow-list" description:"Text file of hostname allow regexes (one per line)"`
-		BindAddress         string        `long:"listen" default:"0.0.0.0:8080" description:"Address:Port to bind to for HTTP"`
+		BindAddress         string        `long:"listen" description:"Address:Port to bind to for HTTP"`
 		BindAddressSSL      string        `long:"ssl-listen" description:"Address:Port to bind to for HTTPS/SSL/TLS"`
 		MaxSize             int64         `long:"max-size" default:"5120" description:"Max allowed response size (KB)"`
 		ReqTimeout          time.Duration `long:"timeout" default:"4s" description:"Upstream request timeout"`
@@ -124,10 +126,6 @@ func main() {
 		mlog.Fatal("HMAC key required")
 	}
 
-	if opts.BindAddress == "" && opts.BindAddressSSL == "" {
-		mlog.Fatal("One of listen or ssl-listen required")
-	}
-
 	if opts.BindAddressSSL != "" && opts.SSLKey == "" {
 		mlog.Fatal("ssl-key is required when specifying ssl-listen")
 	}
@@ -234,6 +232,19 @@ func main() {
 			mlog.Fatal(srv.ListenAndServeTLS(opts.SSLCert, opts.SSLKey))
 		}()
 	}
+	if opts.BindAddress == "" && opts.BindAddressSSL == "" {
+		listeners, err := activation.Listeners()
+		if err != nil {
+			mlog.Fatal("Couldn't find listener sockets for socket activation" +
+					   " and neither listen nor ssl-listen was specified.")
+		}
+		for _, listener := range listeners {
+			go func() {
+				srv := &http.Server{ReadTimeout: 30 * time.Second};
+				mlog.Fatal(srv.Serve(listener))
+			}()
+		}
+	}
 
 	// just block. listen and serve will exit the program if they fail/return
 	// so we just need to block to prevent main from exiting.
